timemachine/md/minimizer.py

```diff
@@ -145,13 +155,20 @@ def minimize_host_4d(mols, host_system, host_coords, ff, box, mol_coords=None) -
     x0 = combined_coords
     v0 = np.zeros_like(x0)
 
-    x0 = fire_minimize(x0, u_impls, box, np.ones(50))
-    # context components: positions, velocities, box, integrator, energy fxns
-    ctxt = custom_ops.Context(x0, v0, box, intg, u_impls)
-    _, xs, _ = ctxt.multiple_steps(np.linspace(1.0, 0, 1000))
+    x0 = fire_minimize(x0, u_impls_1, box, np.ones(50))
+    del u_impls_1
+    x = x0
 
-    final_coords = fire_minimize(xs[-1], u_impls, box, np.zeros(50))
-    for impl in u_impls:
+    for lamb in np.linspace(1.0, 0, 20):
+        u_impls = bind_potentials(make_potentials(lamb))
+        # context components: positions, velocities, box, integrator, energy fxns
+        ctxt = custom_ops.Context(x, v0, box, intg, u_impls)
+        _, xs, _ = ctxt.multiple_steps(lamb * np.ones(50))
+        x = xs[-1]
```

Comment:
Note: this changes the behavior to run for 50 steps at each lambda value (and simulate 1/50th of the original number of windows). The motivation is to amortize the cost of constructing new bound potential objects.